src/main/java/fr/esgi/al5/tayarim/auth/JwtHelper.java

```diff
@@ -0,0 +1,165 @@
+package fr.esgi.al5.tayarim.auth;
+
+import fr.esgi.al5.tayarim.exceptions.TokenExpireOrInvalidException;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.ExpiredJwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import io.jsonwebtoken.security.SignatureException;
+import java.time.Instant;
+import java.util.Date;
+import javax.crypto.SecretKey;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+
+/**
+ * Composant utilitaire pour la gestion des JWT (JSON Web Tokens).
+ */
+@Component
+public class JwtHelper {
+
+  @Value("${PRIVATE_KEY}")
+  private String privateKey;
+
+  /**
+   * Génère un token JWT pour un utilisateur.
+   *
+   * @param id      L'id de l'utilisateur.
+   * @param uuid    L'UUID associé à l'utilisateur.
+   * @param isAdmin Booléen indiquant si l'utilisateur est un administrateur.
+   * @return Le token JWT généré.
+   */
+  public String generateToken(Long id, String uuid, boolean isAdmin) {
+    String subject = id.toString().concat(";").concat(uuid).concat(";").concat(Boolean.toString(isAdmin));
+    var now = Instant.now();
+    return Jwts.builder()
+        .subject(subject)
+        .issuedAt(Date.from(now))
+        .expiration(Date.from(now.plusSeconds(86400))) //24h
+        .signWith(getSignKey())
+        .compact();
+  }
+
+  /**
+   * Extrait le sujet d'un token JWT.
+   *
+   * @param token Le token JWT.
+   * @return Le sujet du token.
+   * @throws TokenExpireOrInvalidException Si le token est expiré ou invalide.
+   */
+  public String extractSubject(String token) {
+    try {
+      Claims tokenBody = getTokenBody(token);
+      if (tokenBody == null) {
+        throw new TokenExpireOrInvalidException();
+      }
+
+      return tokenBody.getSubject();
+    } catch (Exception e) {
+      throw new TokenExpireOrInvalidException();
+    }
+  }
+
+  /**
+   * Extrait l'id de l'utilisateur à partir d'un token JWT.
+   *
+   * @param token Le token JWT.
+   * @return L'id extrait du token.
+   * @throws TokenExpireOrInvalidException Si le token est expiré ou invalide.
+   */
+  public Long extractId(String token) {
+    if (extractSubject(token).split(";")[0] == null) {
+      throw new TokenExpireOrInvalidException();
+    }
+
+    long id = -1L;
+
+    try{
+      id = Long.parseLong(extractSubject(token).split(";")[0]);
+    } catch (Exception e){
+      throw new TokenExpireOrInvalidException();
+    }
+
+    if (id == -1L){
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: '{' is not preceded with whitespace.